internal/cli/build_list.go

```diff
@@ -83,13 +84,26 @@ func (c *BuildListCommand) Run(args []string) int {
 				completeTime = humanize.Time(b.Status.CompleteTime.AsTime())
 			}
 
+			var pipeline string
+			j, err := c.project.Client().GetJob(c.Ctx, &pb.GetJobRequest{
+				JobId: b.JobId,
+			})
+			if err != nil {
+				app.UI.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+				return err
+			}
+			if j.Pipeline != nil {
+				pipeline = j.Pipeline.PipelineName + "[run: " + strconv.FormatUint(j.Pipeline.RunSequence, 10) + "]" + "[step: " + j.Pipeline.Step + "]"
```

Comment:
I think possibly step on its own isn't enough information, we should probably at least include pipeline name + step name? We're also squeezing a decent amount of info into this column, maybe we could separate them by commas and new lines instead like we do with extra data in `waypoint deployment list`? So instead of a single string with `[ ... ]` every item we could do `pipeline: "name",\nstep: "step-name",\nrun: 10` then each row should stretch to fit the new lines and I think a flat list separated by new lines would be easier to read

Reply:
Currently the output looks like:

```
» example-java
    | ID | WORKSPACE | BUILDER |   STARTED   |  COMPLETED  |   PIPELINE                         
----+----+-----------+---------+-------------+-------------+----------------------------
  ✔ | 18 | default   | pack    | 3 hours ago | 3 hours ago | test[run: 2][step: build]  
  ✔ | 17 | default   | pack    | 4 hours ago | 4 hours ago | test[run: 1][step: build]  
```

I don't get what you mean by the line-break option making a flat list, since that would look like this:

```
» example-java
    | ID | WORKSPACE | BUILDER |  STARTED   | COMPLETED  |  PIPELINE    
----+----+-----------+---------+------------+------------+--------------
  ✔ | 18 | default   | pack    | 5 days ago | 5 days ago | test         
     |    |           |         |            |            | run: 2       
     |    |           |         |            |            | step: build  
     ```

Reply:
You can see an example of this in the `Details` field for a `waypoint deployment list`:

```
$ waypoint deployment list 

» example-nodejs
    | ID |  PLATFORM  |               DETAILS               |    STARTED    |   COMPLETED   |      HEALTH
----+----+------------+-------------------------------------+---------------+---------------+--------------------
  ✔ |  1 | kubernetes | image:0624779                       | 8 minutes ago | 8 minutes ago | ✔ - 8 minutes ago
    |    |            | languages:nodejs-engine, nodejs-npm
```

https://github.com/hashicorp/waypoint/blob/ad98496e7bdc2c4fba86d73b1177db65093b9590/internal/cli/deployment_list.go#L232-L250

Since we do it there for extra detailed info, I can see the same format working here pretty well:

```
name:test
run:2
step:build
```

Reply:
Hmm I see what you mean, but I don't think we should have new lines for each field. The current format is also used in `job list`. And the modified `deployment list`:

```
 Waypoint Jobs
              ID             |  OPERATION   |  STATE  | TIME COMPLETED |       TARGET RUNNER        | WORKSPACE |   PROJECT    | APPLICATION  |          PIPELINE           
-----------------------------+--------------+---------+----------------+----------------------------+-----------+--------------+--------------+-----------------------------
 01GBX1K09Q8YAQAZZ62QQNCB5D | Deploy       | Success | 5 days ago     | *                          | default   | example-java | example-java | test[run: 2][step: deploy]  
 01GBX1K09QJP66S0QB0WWRZGYX | Build        | Success | 5 days ago     | *                          | default   | example-java | example-java | test[run: 2][step: build] 
  ```
  
  ```
  » example-java
     | ID | PLATFORM |                                           DETAILS                                            |  STARTED   | COMPLETED  |     HEALTH      
-----+----+----------+----------------------------------------------------------------------------------------------+------------+------------+-----------------
  🚀 | 14 | docker   | image:1231366                                                                                | 5 days ago | 5 days ago | ✔ - 5 days ago  
     |    |          | languages:google.java.runtime, google.java.maven, google.java.entrypoint, google.utils.label  
     |    |          | pipeline: test[run: 2][step: deploy]                                                          
  ✔  | 13 | docker   | image:1231366                                                                                | 5 days ago | 5 days ago | n/a             
     |    |          | languages:google.java.runtime, google.java.maven, google.java.entrypoint, google.utils.label  
     |    |          | pipeline: test[run: 1][step: deploy]     
 ```
 
 If we do a new-line format, then these lists would be really long and ugly:
 
   ```
  » example-java
     | ID | PLATFORM |                                           DETAILS                                            |  STARTED   | COMPLETED  |     HEALTH      
--+----+----------+----------------------------------------------------------------------------------------------+------------+------------+-----------------
  🚀 | 14 | docker   | image:1231366                                                                                | 5 days ago | 5 days ago | ✔ - 5 days ago  
     |     |          | languages:google.java.runtime, google.java.maven, google.java.entrypoint, google.utils.label  
     |     |          | pipeline name: test
     |     |          | pipeline run: 2
     |     |          | pipeline step: deploy                                                         
 ```

Reply:
& the choice of this [run][step] format [came from Jamie](https://docs.google.com/document/d/1Cf3vbdRrxZ539gVxll6HOqEWbhglbmyuFU7osCWk_5o/edit?disco=AAAAc3Czf2Q)